test/integration/vaultdynamicsecret/terraform/postgres.tf

```diff
@@ -69,13 +92,29 @@ resource "vault_database_secret_backend_role" "postgres" {
   ]
 }
 
+resource "vault_database_secret_backend_static_role" "postgres" {
+  namespace           = local.namespace
+  backend             = vault_database_secrets_mount.db.path
+  name                = local.db_role_static
+  db_name             = vault_database_secrets_mount.db.postgresql[0].name
+  username            = local.db_role_static_user
+  rotation_statements = ["ALTER USER \"{{name}}\" WITH PASSWORD '{{password}}';"]
+  rotation_period     = 30
+  depends_on = [
+    null_resource.create-pg-user,
```

Comment:
this looks odd, I guess I'm learning something new today :)
Is this primarily due to a lack of being able to create a static role in tf for the db engine?

Reply:
A pre-existing PG "user" must be configured before we can provision the vault static role. The use of a `depends_on` is needed since a `null_resource` has no result that could be used by the dependent resource.